core/db/migrate/20160420181916_migrate_credit_cards_to_wallet_payment_sources.rb

```diff
@@ -1,15 +1,23 @@
-class MoveCreditCardDefaultToWalletPaymentSource < ActiveRecord::Migration[4.2]
+class MigrateCreditCardsToWalletPaymentSources < ActiveRecord::Migration[4.2]
+  class CreditCard < ActiveRecord::Base
+    self.table_name = 'spree_credit_cards'
+  end
+  class WalletPaymentSource < ActiveRecord::Base
+    self.table_name = 'spree_wallet_payment_sources'
+  end
+
   def up
     credit_cards = Spree::CreditCard.
-      where(default: true).
+      where("gateway_customer_profile_id IS NOT NULL").
       where.not(user_id: nil)
 
     credit_cards.find_each do |credit_card|
-      Spree::WalletPaymentSource.create!(
+      Spree::WalletPaymentSource.find_or_create_by!(
         user_id: credit_card.user_id,
-        payment_source: credit_card,
-        default: credit_card.read_attribute(:default),
-      )
+        payment_source: credit_card
+      ) do |wallet_source|
+        wallet_source.default = credit_card[:default]
```

Comment:
Passing a block to `find_or_create_by` behaves the same as passing a block to `create` if the record doesn't already exist.

We want to do this here, because we absolutely only want to have one `WalletPaymentSource` per credit card, the block specifies the values if one doesn't already exist. This makes the up migration idempotent, which we want since the down migration doesn't remove `WalletPaymentSource`s.